spec/lib/pubmed/query_author_spec.rb

```diff
@@ -195,4 +167,70 @@
+    let(:query_blank_author) { described_class.new(blank_author, options) }
+    let(:query_or_author) { described_class.new(or_author, options) }
+    let(:query_not_author) { described_class.new(not_author, options) }
+    let(:query_nil_author_first_name) { described_class.new(nil_author_first_name, options) }
+    let(:query_nil_author_last_name) { described_class.new(nil_author_last_name, options) }
+    let(:query_author_only_one_bad_name) { described_class.new(author_one_bad_name, options) }
+
+    let(:space_author) { create :author, :space_first_name }
+    let(:period_author) { create :author, :period_first_name }
+    let(:blank_author) { create :author, :blank_first_name }
+    let(:or_author) { create :author, :or_first_name }
+    let(:not_author) { create :author, :not_last_name }
+    let(:nil_author_first_name) { create :author, :nil_first_name }
+    let(:nil_author_last_name) { create :author, :nil_last_name }
+    let(:author_one_bad_name) { create :author_with_alternate_identities, :or_first_name }
+
+    context 'with a user with valid first names' do
+      it 'indicates it is a valid query' do
+        expect(query_author).to be_valid
+      end
+    end
+
+    context 'with a user with one invalid name and a second valid name' do
+      it 'indicates it is a valid query, ignoring the bad name in the query' do
+        expect(query_author_only_one_bad_name).to be_valid
+        expect(query_author_only_one_bad_name.send(:term)).to eq('((Edler1, Alice1[Author])) AND (Stanford University[Affiliation] OR Example University[Affiliation])')
+      end
+    end
+
+    context 'with a user with no valid names' do
+      it 'indicates that name with a period for a first name is not a valid query' do
+        expect(query_period_author).not_to be_valid
+      end
+
+      it 'indicates that a name with a space for a first name is not a valid query' do
+        expect(query_space_author).not_to be_valid
+      end
+
+      it 'indicates that a name with a blank for a first name is not a valid query and returns no pmids' do
+        expect(query_blank_author).not_to be_valid
+        expect(query_blank_author.pmids).to be_empty
+      end
+
+      it 'indicates that a name with nil as a first name is not a valid query' do
+        expect(query_nil_author_first_name).not_to be_valid
+      end
+
+      it 'indicates that a name with nil as a last name is not a valid query' do
+        expect(query_nil_author_last_name).not_to be_valid
+      end
+
+      it 'indicates that a name with "Or" as a first name is not a valid query and returns no pmids' do
+        expect(query_or_author).not_to be_valid
+        expect(query_or_author.pmids).to be_empty
+      end
+
+      it 'indicates that a name with "Not" as a last name is not a valid query and returns no pmids' do
+        expect(query_not_author).not_to be_valid
+        expect(query_not_author.pmids).to be_empty
+      end
+    end
+  end
 end
```

Comment:
no tests were eliminated here, just moved to a describe block, along with adding new tests verifying that names that include "Or" and "Not" are ignored, as well as nil last names